src/main/kotlin/com/jetbrains/micropython/nova/FileSystemWidget.kt

```diff
@@ -246,8 +243,18 @@ class FileSystemWidget(val project: Project, newDisposable: Disposable) :
     }
 
     @Throws(IOException::class)
-    suspend fun upload(relativeName: @NonNls String, contentsToByteArray: ByteArray) =
-        comm.upload(relativeName, contentsToByteArray)
+    suspend fun upload(relativeName: @NonNls String, contentsToByteArray: ByteArray) {
+        try {
+            comm.upload(relativeName, contentsToByteArray)
+        } catch (e: CancellationException) {
+            withContext(NonCancellable) {
+                withContext(Dispatchers.EDT) {
+                    refresh()
```

Comment:
This approach is not correct. 
1. `withContext(Dispatchers.EDT)` leads to freezes here. EDT code inside `refresh()`  is already under the context
1. It's better to avoid `NonCancellable` context because it may freeze UI for LONG_TIMEOUT. Refresh action should be cancellable, and if `refresh` cancelled, everything is already going wrong, and it's better to disconnect from the board.

Reply:
Probably that refresh cancellation code should be inside `refresh()`

Reply:
I have removed the unnecessary `withContext(Dispatchers.EDT)` calls. 

And for the `NonCancellable` context. The reason I used `NonCancellable` is that without it the refresh simply won't happen. The progress bar's cancellation will also propagate to the `refresh()` method call in the `CancellationException` handler for deletion and upload methods. Using `NonCancellable` worked for me, without having to modify many more lines of code.

There might be a better way to handle this, I am not too familiar with plugin development, let alone Kotlin. So please tell me if this can be handled better differently.

Reply:
Sure, I'll take care of this cancellability